py/desisurvey/NTS.py

```diff
@@ -0,0 +1,200 @@
+            ind, mask = self.scheduler.tiles.index(previoustiles,
+                                                   return_mask=True)
+            self.scheduler.in_night_pool[ind[mask]] = False
+        # remove previous tiles from possible tiles to schedule
+        # note: this will be remembered until NTS is restarted!  EFS
+        result = self.scheduler.next_tile(
+            mjd, self.ETC, seeing, transparency, skylevel, program=program)
+        (tileid, passnum, snr2frac_start, exposure_factor, airmass,
+         sched_program, mjd_program_end) = result
+        if tileid is None:
+            return {'tileid': None, 's2n': 0., 'esttime': 0., 'maxtime': 0.,
+                    'fiber_assign': '', 'foundtile': False}
+
+        # lastexp ignored, fiberassign ignored.  EFS
+        texp_tot, texp_remaining, nexp_remaining = self.ETC.estimate_exposure(
+            sched_program, snr2frac_start, exposure_factor, nexp_completed=0)
+
+        # s2n: this is really what we should be passing back, but currently
+        # scheduler thinks in terms of texp in 'nominal' conditions.  Want
+        # guidance converting this to s2n...
+        # what cosmic split related elements should I be thinking about here?
+
+        s2n = 50.0 * texp_remaining/(
+            self.ETC.TEXP_TOTAL[sched_program]*exposure_factor)  # EFS hack
+        exptime = texp_remaining
+        maxtime = self.ETC.MAX_EXPTIME
+        self.scheduler.update_snr(
+            tileid, snr2frac_start + min([exptime, maxtime])/texp_tot)
+        self.scheduler.save('scheduler_{}.fits'.format(self.night.isoformat()))
+        if program is None:
+            maxtime = min([maxtime, mjd_program_end-maxtime])
+
+        fiber_assign = os.path.join(self.fiber_assign_dir,
+                                    'tile_%d.fits' % tileid)
+        days_to_seconds = 60*60*24
+
+        selection = {'tileid': tileid, 's2n': s2n,
+                     'esttime': exptime*days_to_seconds,
+                     'maxtime': maxtime*days_to_seconds,
+                     'fiber_assign': fiber_assign,
+                     'foundtile': True}
+
+        return selection
+
+
+def afternoon_plan(night=None, lastnight=None):
+    if night is None:
+        night = datetime.date.today().isoformat()
+    rules = desisurvey.rules.Rules()
+    # should look for rules file in obsplan dir?
+    if lastnight is not None:
+        planner = desisurvey.plan.Planner(
+            rules, restore='planner_end_{}.fits' % lastnight)
+        scheduler = desisurvey.scheduler.Scheduler(
+            restore='scheduler_end_{}.fits')
+    else:
+        planner = desisurvey.plan.Planner(rules)
+        scheduler = desisurvey.scheduler.Scheduler()
+    # restore: maybe check directory, and restore if file present?  EFS
+    # planner.save(), scheduler.save()
+    # planner.restore(), scheduler.restore()
+    import dateutil.parser
+    planner.afternoon_plan(dateutil.parser.parse(night).date(),
```

Comment:
I suggest using `desisurvey.utils.get_date(night)` here so we only have one entry point for night -> date conversion.

Reply:
Should be resolved in latest commits.